src/plugins/data/server/query/routes.ts

```diff
@@ -13,12 +13,15 @@ import { SavedQueryRouteHandlerContext } from './route_handler_context';
 import { SavedQueryRestResponse } from './route_types';
 import { SAVED_QUERY_BASE_URL } from '../../common/constants';
 
+export const SAVED_QUERY_NAME_MAX_LENGTH = 50;
+export const SAVED_QUERY_DESCRIPTION_MAX_LENGTH = 200;
```

Comment:
One thing we should consider is if this is a breaking change. The selected sizes seem reasonable IMO, but we don't really know how users are using these fields today. For example a user might rely on the description field to store a detailed description of why a query exists and how it can be used, which could easily exceed 200 characters. After upgrading, if they were to modify this query in some way, they wouldn't be able to save it again without truncating the description.

I wonder if there's a way (maybe through some config) to support this without breaking existing use cases? Addressing this could also help set the groundwork for a more holistic approach to fixing the issue across SO types longer term. Maybe we'll decide it's fine to merge as is, but I think it's worth considering and discussing first.

Reply:
even worse, like @stratoula discovered, this would even break the load query functionality due to output validation ... so it's a handle with care situation.

Reply:
Yeah I am a bit sceptical about this change 🤔 Maybe the safest thing for now is to only have a UI validation?

Reply:
Output validation is only enabled in dev (to help us catch these issues) so users wouldn't encounter this, but I agree the schema's an issue regardless.